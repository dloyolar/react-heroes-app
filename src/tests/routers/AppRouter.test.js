import { mount } from 'enzyme';
import { AuthContext } from '../../auth/authContext';
import { AppRouter } from '../../routers/AppRouter';

describe('Test on <AppRouter />', () => {
  test('should be show login screen if isnt auth', () => {
    const contextValue = {
      user: {
        logged: false,
      },
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text().trim()).toBe('Login');
  });

  test('should be show marvel screen if is auth', () => {
    const contextValue = {
      user: {
        logged: true,
        name: 'Diego',
      },
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.navbar').exists()).toBe(true);
  });
});
