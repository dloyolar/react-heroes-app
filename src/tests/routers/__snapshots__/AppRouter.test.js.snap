// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AppRouter /> should be show login screen if isnt auth 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <p
    className="text-muted"
  >
    (Don't worry, this is a fake login, you dont need any user or credentials ðŸ‘Œ)
  </p>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;

exports[`Test on <AppRouter /> should be show marvel screen if is auth 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link "
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link "
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Diego
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        MarvelScreen
      </h1>
      <hr />
      <div
        className="row row-cols-1 row-cols-md-3 g-4 animate__animated animate__fadeIn"
      >
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-spider"
          onClick={[Function]}
        >
          <img
            alt="Spider Man"
            className="img img-responsive"
            src="/assets/marvel-spider.jpg"
          />
          <div
            className="profile-name"
          >
            Spider Man
          </div>
          <div
            className="profile-position"
          >
            Peter Parker
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p
                    className="fw-bold"
                  >
                    First appaerance: 
                    <br />
                    Amazing Fantasy #15
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-captain"
          onClick={[Function]}
        >
          <img
            alt="Captain America"
            className="img img-responsive"
            src="/assets/marvel-captain.jpg"
          />
          <div
            className="profile-name"
          >
            Captain America
          </div>
          <div
            className="profile-position"
          >
            Steve Rogers
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p
                    className="fw-bold"
                  >
                    First appaerance: 
                    <br />
                    Captain America Comics #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-iron"
          onClick={[Function]}
        >
          <img
            alt="Iron Man"
            className="img img-responsive"
            src="/assets/marvel-iron.jpg"
          />
          <div
            className="profile-name"
          >
            Iron Man
          </div>
          <div
            className="profile-position"
          >
            Tony Stark
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p
                    className="fw-bold"
                  >
                    First appaerance: 
                    <br />
                    Tales of Suspense #39
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-thor"
          onClick={[Function]}
        >
          <img
            alt="Thor"
            className="img img-responsive"
            src="/assets/marvel-thor.jpg"
          />
          <div
            className="profile-name"
          >
            Thor
          </div>
          <div
            className="profile-position"
          >
            Thor Odinson
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p
                    className="fw-bold"
                  >
                    First appaerance: 
                    <br />
                    Journey into Myster #83
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-hulk"
          onClick={[Function]}
        >
          <img
            alt="Hulk"
            className="img img-responsive"
            src="/assets/marvel-hulk.jpg"
          />
          <div
            className="profile-name"
          >
            Hulk
          </div>
          <div
            className="profile-position"
          >
            Bruce Banner
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p
                    className="fw-bold"
                  >
                    First appaerance: 
                    <br />
                    The Incredible Hulk #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-wolverine"
          onClick={[Function]}
        >
          <img
            alt="Wolverine"
            className="img img-responsive"
            src="/assets/marvel-wolverine.jpg"
          />
          <div
            className="profile-name"
          >
            Wolverine
          </div>
          <div
            className="profile-position"
          >
            James Howlett
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p
                    className="fw-bold"
                  >
                    First appaerance: 
                    <br />
                    The Incredible Hulk #180
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-daredevil"
          onClick={[Function]}
        >
          <img
            alt="Daredevil"
            className="img img-responsive"
            src="/assets/marvel-daredevil.jpg"
          />
          <div
            className="profile-name"
          >
            Daredevil
          </div>
          <div
            className="profile-position"
          >
            Matthew Michael Murdock
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p
                    className="fw-bold"
                  >
                    First appaerance: 
                    <br />
                    Daredevil #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-hawkeye"
          onClick={[Function]}
        >
          <img
            alt="Hawkeye"
            className="img img-responsive"
            src="/assets/marvel-hawkeye.jpg"
          />
          <div
            className="profile-name"
          >
            Hawkeye
          </div>
          <div
            className="profile-position"
          >
            Clinton Francis Barton
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p
                    className="fw-bold"
                  >
                    First appaerance: 
                    <br />
                    Tales of Suspense #57
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-cyclops"
          onClick={[Function]}
        >
          <img
            alt="Cyclops"
            className="img img-responsive"
            src="/assets/marvel-cyclops.jpg"
          />
          <div
            className="profile-name"
          >
            Cyclops
          </div>
          <div
            className="profile-position"
          >
            Scott Summers
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p
                    className="fw-bold"
                  >
                    First appaerance: 
                    <br />
                    X-Men #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-silver"
          onClick={[Function]}
        >
          <img
            alt="Silver Surfer"
            className="img img-responsive"
            src="/assets/marvel-silver.jpg"
          />
          <div
            className="profile-name"
          >
            Silver Surfer
          </div>
          <div
            className="profile-position"
          >
            Norrin Radd
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p
                    className="fw-bold"
                  >
                    First appaerance: 
                    <br />
                    The Fantastic Four #48
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>,
]
`;
